TypeCobol/Compiler/Domain/SymbolTableBuilder.cs

```diff
@@ -0,0 +1,190 @@
+        /// </summary>
+        public static TypeCobolConfiguration Config
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Any Base Cobol Symbol table.
+        /// </summary>
+        public static SymbolTable BaseTable
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Path of the Intrinsic file
+        /// </summary>
+        public static string IntrinsicPath
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Called when A node has been syntactically recognized by the TypeCobol Parser.
+        /// </summary>
+        /// <param name="node">The node being built</param>
+        /// <param name="program">The Program that contains the node.</param>
+        public abstract override void OnNode(Node node, Program program);
+
+        /// <summary>
+        /// The Root Symbol Table.
+        /// </summary>
+        internal static RootSymbolTable _rootSymbolTable;
+
+        /// <summary>
+        /// The Root Symbol Table
+        /// </summary>
+        public static RootSymbolTable Root
+        {
+            get
+            {
+                if (_rootSymbolTable == null)
+                {
+                    _rootSymbolTable = new RootSymbolTable();
+                    //Store Builtin Symbols.
+                    BuiltinSymbols.StoreSymbols(_rootSymbolTable.Types);
+                    LoadBaseTable();
+                }
+                return _rootSymbolTable;
+            }
+            set { _rootSymbolTable = value; }
+        }
+
+        /// <summary>
+        /// Load intrinsic symbol in the global table.
+        /// </summary>
+        /// <param name="path"></param>
+        public static void LoadIntrinsics(string path)
+        {
+            path = path ?? IntrinsicPath;
+            if(path == null)
+            {
+                return; 
+            }
+            //Allocate a static Program Symbol Table Builder
+            ProgramSymbolTableBuilder Builder = null;
+            NodeListenerFactory BuilderNodeListenerFactory = () =>
+            {
+                Builder = new ProgramSymbolTableBuilder();
+                return Builder;
+            };
+            Compiler.Parser.NodeDispatcher.RegisterStaticNodeListenerFactory(BuilderNodeListenerFactory);
+            //Put all Types from programs into the Global Namespace.
+            TransfertAllProgramsToIntrinsics();
+            try
+            {
+                //path, skeletons, DocumentFormat.RDZReferenceFormat, typeCobolVersion, autoRemarks, copies
+                //ParseGenerateCompare(string path, List<Skeleton> skeletons = null, bool autoRemarks = false, string typeCobolVersion = null, IList<string> copies = null) {
+                TypeCobol.Parser parser = TypeCobol.Parser.Parse(path, DocumentFormat.RDZReferenceFormat, /*bool autoRemarks =*/ false, /*IList<string> copies =*/ null);
```

Comment:
Can you add a comment: "There is no COPY in our intrinsic, so don't set AutoRemarks and copyFolders".

Reply:
ok